### PR TITLE
(VANAGON-79) Use VanagonLogger to classify output.

### PR DESCRIPTION
**Pull request by jackie-kinsler**
Use VanagonLogger (inherits from Logger class) to classify output. 

Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

**Review by e-gris — COMMENTED** [review]
It would be my preference to avoid class methods and use instance methods instead.

Now, logger classes are the classic example for singleton classes so I feel slightly conflicted in my request and would easily be convinced otherwise. I want to see if there's any motivation for going with the class method for logging.

File: lib/vanagon/component/source/rewrite.rb [hunk 1]
```diff
@@ -16,7 +17,7 @@ class << self
           # @deprecated Please use the component DSL method #mirror(<URI>)
           #   instead. This method will be removed before Vanagon 1.0.0.
           def register_rewrite_rule(protocol, rule)
-            warn <<-HERE.undent
+            VanagonLogger.info <<-HERE.undent
```
Comment (e-gris) [inline lib/vanagon/component/source/rewrite.rb:20]: Should be 'warn'

File: lib/vanagon/engine/always_be_scheduling.rb [hunk 2]
```diff
@@ -245,10 +246,10 @@ def select_target_from(pooler) # rubocop:disable Metrics/AbcSize
         )
 
         unless response.code == "202"
-          warn "failed to request ABS with code #{response.code}"
+          VanagonLogger.info "failed to request ABS with code #{response.code}"
```
Comment (e-gris) [inline lib/vanagon/engine/always_be_scheduling.rb:249]: should be 'warn'

File: lib/vanagon/engine/always_be_scheduling.rb [hunk 3]
```diff
@@ -245,10 +246,10 @@ def select_target_from(pooler) # rubocop:disable Metrics/AbcSize
...
           if valid_json?(response.body)
             response_json = JSON.parse(response.body)
-            warn "reason: #{response_json['reason']}"
+            VanagonLogger.info "reason: #{response_json['reason']}"
```
Comment (e-gris) [inline lib/vanagon/engine/always_be_scheduling.rb:252]: should be 'warn'

File: lib/vanagon/environment.rb [hunk 4]
```diff
@@ -147,7 +148,7 @@ def sanitize_subshells(str)
         update your project's parameters.
       WARNING
 
-      warn warning.join("\n")
+      VanagonLogger.info warning.join("\n")
```
Comment (e-gris) [inline lib/vanagon/environment.rb:151]: should be 'warn'

File: lib/vanagon/platform.rb [hunk 5]
```diff
@@ -401,7 +402,7 @@ def is_cisco_wrlinux?
     # @deprecated Please use is_macos? instead
     # @return [true, false] true if it is an osx variety, false otherwise
     def is_osx?
-      warn "is_osx? is a deprecated method, please use #is_macos? instead."
+      VanagonLogger.info "is_osx? is a deprecated method, please use #is_macos? instead."
```
Comment (e-gris) [inline lib/vanagon/platform.rb:405]: should be 'warn'

File: lib/vanagon/platform.rb [hunk 6]
```diff
@@ -539,7 +540,7 @@ def version_munger(version_string, default: '=')
       match = version_string.match(VERSION_REGEX)
 
       if match.nil?
-        warn "Passing a version without an operator is deprecated!"
+        VanagonLogger.info "Passing a version without an operator is deprecated!"
```
Comment (e-gris) [inline lib/vanagon/platform.rb:543]: should be 'warn'

File: lib/vanagon/platform/dsl.rb [hunk 7]
```diff
@@ -259,7 +260,7 @@ def vmpooler_template(template_name)
       # @param name [String] name that the pooler uses for this platform
       # @deprecated Please use vmpooler_template instead, this will be removed in a future vanagon release.
       def vcloud_name(cloud_name)
-        warn "vcloud_name is a deprecated platform DSL method, and will be removed in a future vanagon release. Please use vmpooler_template instead."
+        VanagonLogger.info "vcloud_name is a deprecated platform DSL method, and will be removed in a future vanagon release. Please use vmpooler_template instead."
```
Comment (e-gris) [inline lib/vanagon/platform/dsl.rb:263]: More 'warns'

File: spec/lib/vanagon/cli_spec.rb [hunk 8]
```diff
@@ -8,7 +9,7 @@
     begin
       ex.run
     rescue SystemExit => e
-      puts "Got SystemExit: #{e.inspect}. Ignoring"
+      VanagonLogger.error "Got SystemExit: #{e.inspect}. Ignoring"
```
Comment (e-gris) [inline spec/lib/vanagon/cli_spec.rb:12]: This should stay 'puts'

File: spec/lib/vanagon/engine/ec2_spec.rb [hunk 9]
```diff
@@ -1,7 +1,9 @@
+require 'vanagon/logger'
+
 begin
   require 'aws-sdk'
 rescue LoadError
-  $stderr.puts "Unable to load AWS SDK; skipping optional EC2 engine spec tests"
+  VanagonLogger.error "Unable to load AWS SDK; skipping optional EC2 engine spec tests"
```
Comment (e-gris) [inline spec/lib/vanagon/engine/ec2_spec.rb:6]: This should stay 'puts'

**Review by e-gris — COMMENTED** [review]
Lots of little comments.

Mostly, I'm confused why a new VanagonLogger class with reimplented methods. Why not create a module with the formatter code and put that in the object instance?

tefoji (https://github.com/puppetlabs/tefoji/) has an example of what I'm thinking but want to be cautious because there may not be complete translation between the two cases.

File: lib/vanagon/cli.rb [hunk 10]
```diff
@@ -63,10 +65,10 @@ def parse(argv) # rubocop:disable Metrics/AbcSize
       when 'ship'
         @sub_parser = Vanagon::CLI::Ship.new
       when 'help'
-        puts DOCUMENTATION
+        VanagonLogger.warn DOCUMENTATION
```
[the remark this answers is not in the document]
Comment (jackie-kinsler) [reply lib/vanagon/cli.rb:68]: I like that -- should never be dependent on what the logger level is.

**Comment by jackie-kinsler** [comment]
NOTE: Much of the logging that should seemingly be `VanagonLogger.warn` is set to `VanagonLogger.info` to prevent things being put to STDOUT. Some tools depend on Vanagon having very limited STDOUT, and `VanagonLogger.info` goes to STDERR rather than STDOUT.

**Pull request merged**